src/main/java/pl/matidominati/GitHubApp/service/RepositoryService.java

```diff
@@ -0,0 +1,92 @@
+        return mapper.mapRepositoryDetailsToResponseDto(localRepository);
+    }
+
+    public RepositoryResponseDto getRepository(String owner, String repositoryName) {
+        var localRepository = repository.findByOwnerUsernameAndRepositoryName(owner, repositoryName)
+                .orElseThrow(() -> new DataNotFoundException("Incorrect username or repository name provided."));
+        return mapper.mapRepositoryDetailsToResponseDto(localRepository);
+    }
+
+    @Transactional
+    public RepositoryResponseDto editRepository(String owner, String repositoryName, RepositoryPojo updatedRepository) {
+        var originalRepository = getDetails(owner, repositoryName);
+        updateRepositoryDetails(originalRepository, updatedRepository);
+        var repoToSave = mapper.mapPojoToRepositoryDetails(originalRepository);
+        repository.save(repoToSave);
+        return mapper.mapRepositoryDetailsToResponseDto(repoToSave);
+    }
+
+    @Transactional
+    public void deleteRepository(String owner, String repositoryName) {
+        RepositoryDetails repositoryToDelete = repository.findByOwnerUsernameAndRepositoryName(owner, repositoryName)
+                .orElseThrow(() -> new DataNotFoundException("Incorrect username or repository name provided."));
+        repository.delete(repositoryToDelete);
+    }
+
+    public RepositoryPojo getDetails(String owner, String repositoryName) {
+        var localRepository = repository.findByOwnerUsernameAndRepositoryName(owner, repositoryName)
+                .orElseThrow(() -> new DataNotFoundException("Incorrect username or repository name provided."));
+        return mapper.mapRepositoryDetailsToPojo(localRepository);
+    }
+
+    public static void updateRepositoryDetails(RepositoryPojo originalRepository, RepositoryPojo updatedRepository) {
+        if (updatedRepository.getFullName() != null && !updatedRepository.getFullName().equals(originalRepository.getFullName())) {
+            originalRepository.setFullName(updatedRepository.getFullName());
+        }
+        if (updatedRepository.getDescription() != null && !updatedRepository.getDescription().equals(originalRepository.getDescription())) {
+            originalRepository.setDescription(updatedRepository.getDescription());
+        }
+        if (updatedRepository.getCloneUrl() != null && !updatedRepository.getCloneUrl().equals(originalRepository.getCloneUrl())) {
+            originalRepository.setCloneUrl(updatedRepository.getCloneUrl());
+        }
+        if (updatedRepository.getStars() != originalRepository.getStars()) {
+            originalRepository.setStars(updatedRepository.getStars());
+        }
+        if (updatedRepository.getCreatedAt() != null && !updatedRepository.getCreatedAt().isAfter(LocalDateTime.now())) {
+            originalRepository.setCreatedAt(updatedRepository.getCreatedAt());
+        }
+    }
+
+    public static RepositoryDetails createRepository(RepositoryPojo repositoryPojo) {
```

Comment:
to taka rzecz co srednio zeby byla w serwisie. Lepiej wystawic jakis nie wiem RepositoryFactory nowa klase czy cos ktora bedzie od tego